library/qtest/src/Operations.qs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/// # Summary
+/// Runs a series of named test operations with expected results.
+///
+/// # Description
+/// Given an array of test names, test operations, and expected results (of the same type), returns `true`
+/// if all tests passed, and `false` if any one test failed.
+/// Prints the expected and received result for any failed tests.
+///
+/// # Input
+/// ## test_cases
+/// An array of five-arity tuples of the form `(test_name, num_qubits, qubit_prep_callable, callable_to_test, expected_result)`.
+/// `num_qubits` will be allocated in a qubit array and passed to `qubit_prep_callable` to prepare the state before executing 
+/// `callable_to_test`. Afterwards, `callable_to_test` will be called and its result will be compared to `expected_result`.
+///
+/// # Example
+/// ```qsharp
+/// function Main() : Unit {
+///     TestCases([
+///         ("Should return 42", TestCaseOne, 42),
+///         ("Should add one", () => AddOne(5), 6)
+///     ]);
+/// }
+///
+/// function TestCaseOne() : Int {
+///     42
+/// }
+///
+/// function AddOne(x: Int) : Int {
+///     x + 1
+/// }
+/// ```
+operation TestCases<'Result : Eq + Show > (test_cases : (String, Int, (Qubit[]) => (), (Qubit[]) => 'Result, 'Result)[]) : Bool {
+    let failed_test_buf = TestCasesSilent(test_cases);
+    
+    if Length(failed_test_buf) == 0 {
+        Message($"{Length(test_cases)} test(s) passed.");
```

Comment:
We could simplify this by saying `Message($"{Length(test_cases)}-Length(failed_test_buf)}/{Length(test_cases)} test(s) passed."); outside of if condition.

Reply:
Updated the input to provide some more verbose output.